src/models/CompanyApplication.js

```diff
@@ -150,6 +161,13 @@ export const isRejectable = (application) => {
     return true;
 };
 
+
+CompanyApplicationSchema.methods.companyValidation = function() {
+    if (this.isVerified) throw new APIError(HTTPStatus.FORBIDDEN, ErrorTypes.FORBIDDEN, ValidationReasons.ALREADY_VALIDATED);
+    this.isVerified = true;
+    return this.save({ validateModifiedOnly: true });
+};
```

Comment:
I don't think we should be throwing an APIError at the model level of our platform. Following the other patterns, it should simply be an error, if I'm not mistaken

Reply:
You're right: the way this method is implemented ties our data layer with the route layer, which can pose more work should we want to change either one of them.